### ISSUE TITLE
UI: Add metadata to UI for Jobs and other entities that support metadata

### ISSUE THREAD

**Issue by DerekStrickland** [hashicorp/nomad#11131]
### Proposal

Nomad allows users to add metadata to Jobs, Tasks, and other Nomad entities, but that metadata is not easily discoverable. You can look at the Job Definition for instance, but only if you know what you are looking for. It would be nice if you could see metadata about a Job, Task, etc immediately when viewing jobs in a list or on the detail screen.  A secondary nice to have would be the ability to filter by metadata to find specific jobs. etc or entities of any type that share a metadata element.  

### Use-cases

The end goal would be to either 

A) Provide at a glance context about Jobs, Tasks, etc to users reviewing lists or details about items, or
B) Allow filtering out irrelevant items when trying to assess a set of related or similar items

**Comment by ChaiWithJai** [comment]
Marking this into the `backlog` after the office hours discussion. I think this would require some design exploration (and product roadmapping?) before we can make this a reality.

cc:  @joshklekamp @mikenomitch @philrenaud @juliezzhou
